sphinx_gallery/notebook.py

```diff
@@ -364,10 +365,22 @@ def python_to_jupyter_cli(args=None, namespace=None):
                         ' (e.g. *.py)')
     args = parser.parse_args(args, namespace)
 
+    # handle `sphinx_gallery_conf`
+    defaults = copy.deepcopy(gen_gallery.DEFAULT_GALLERY_CONF)
+    if sphinx_gallery_conf is None:
+        sphinx_gallery_conf = defaults
+    else:
+        if not isinstance(sphinx_gallery_conf, dict):
+            raise TypeError("`sphinx_gallery_conf` must be dict, "
+                            "got %s" % type(sphinx_gallery_conf))
+        for name in defaults:
+            if name not in sphinx_gallery_conf:
+                sphinx_gallery_conf[name] = defaults[name]
```

Comment:
A simpler way would be:
```suggestion
    gallery_conf = copy.deepcopy(gen_gallery.DEFAULT_GALLERY_CONF)
    if sphinx_gallery_conf is not None:
        gallery_conf.update(sphinx_gallery_conf)
```
then pass `gallery_conf` below instead of `sphinx_gallery_conf`.

It loses the explicit `if not isinstance` but presumably whatever `TypeError` the native `dict` gives for a bad `update` param is good enough. And we get the benefit of supporting whatever `update` does, including key-value pairs in iterable for example.